text/0028-bill-of-materials.md

```diff
@@ -0,0 +1,166 @@
+[`buildpack.toml` file](https://github.com/paketo-buildpacks/node-engine/blob/main/buildpack.toml).
+
+The minimal set of keys to include in these type of BOM entries are:
+```
+[[bom]]
+name = "<dependency name>"
+
+[bom.metadata]
+  sha256 = "<hash of dependency artifact from uri>"
+  uri = "<uri to dependency>"
+  version = "<dependency version>"
+```
+The optional set of keys may include:
+```
+  # Optional parameters
+[bom.metadata]
+  cpe = "<version-specific common platfrom enumeration>"
+  purl = "<package URL>"
+  licenses = [<licenses that the dependency has>]
+
+[bom.metadata.source]
+  uri = "<uri to the dependency source>"
+  sha256 = "<hash of the dependency source artifact from source-uri>"
+```
+
+### Indirectly Installed Dependencies
+The final component that we should aim to publish BOM metadata for is for
+dependencies that are indirectly installed. These types of dependencies are
+either downloaded during the image building process or vendored as part of the
+application. If Angular was installed as a module by the [Paketo NPM Install
+Buildpack](https://github.com/paketo-buildpacks/npm-install), it would fall
+under this category as something we'd want to provide BOM metadata for.
+
+The BOM entries for this category should also include information about the
+modules available in the final image, as well as those used to construct the
+image.
+
+The minimal set of keys to include in package module BOM entries are:
+```
+[[bom]]
+name = "<module name>"
+
+[bom.metadata]
+  version = "<module version>"
+```
+Note that this should have the same structure as the runtime and compilation
+dependency BOM entries. Some fields (such as `uri` , for example) have been
+omitted until further investigation is done to find out how these can be
+obtained. The final set of fields will be some subset of the fields available
+on our Directly Installed Dependencies.
+
+### License Information
+
+As a note, the plan is to use various scanning tools in order to obtain license
+information for all dependencies. It should be assumed that you are receiving
+all of the output from a given license scanning tool with no further vetting at
+this time. If this could potentially cause issues for your compliance process
+you should go through with any advanced compliance that is normally required.
+However, there should be a good faith effort to provide only false-postive
+license results to ensure that individuals can still trust that all relevant
+licenses are present. This level of confidence and verification may change in
+the future, but that change should be communicated in later documentation.
+
+### Package URLs
+
+[Package URLs](https://github.com/package-url/purl-spec) will be provided for
+our directly installed dependencies. These types of dependencies (runtime
+dependencies) do not have explicitly supported types in the [Package URL
+Types](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst)
```

Comment:
```suggestion
our directly installed dependencies. Some of these dependencies (runtime
dependencies) do not have explicitly supported types in the [Package URL
Types](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst)
```

There are some cases where dependencies have defined PURL types. For example, the [spring-cloud-bindings dependency](https://github.com/paketo-buildpacks/spring-boot/blob/main/buildpack.toml#L48) installed by paketo-buildpacks/spring-boot, would be type `maven`.